File: puzzles/go/shadows-knight1.go
package main

import "fmt"

func main() {
  var width, height int
  fmt.Scan(&width, &height)
  
  var n int // maximum number of turns before game over.
  fmt.Scan(&n)
  
  var x, y int
  fmt.Scan(&x, &y)
  
  var minX = 0
  var maxX = width - 1
  var minY = 0
  var maxY = height - 1

  for {
    var bombDir string
    fmt.Scan(&bombDir)
    
    switch bombDir {
    case "U":
      minX, maxX = x, x
      maxY = y - 1
    case "UR":
      minX = x + 1
      maxY = y - 1
    case "R":
      minY, maxY = y, y
      minX = x + 1
    case "DR":
      minX = x + 1
      minY = y + 1
    case "D":
      minX, maxX = x, x
      minY = y + 1
    case "DL":
      maxX = x - 1
      minY = y + 1
    case "L":
      minY, maxY = y, y
      maxX = x - 1
    case "UL":
      maxX = x - 1
      maxY = y - 1
    }

    x = (minX + maxX) / 2
    y = (minY + maxY) / 2
    
    fmt.Printf("%v %v\n", x, y) // the location of the next window Batman should jump to.
  }
}

File: puzzles/cpp/shadows-knight1.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
  int width; // width of the building.
  int height; // height of the building.
  cin >> width >> height; cin.ignore();
  int n;
  cin >> n; cin.ignore();
  int x, y;
  cin >> x >> y; cin.ignore();
  int minX = 0;
  int maxX = width - 1;
  int minY = 0;
  int maxY = height - 1;

  // game loop
  while (true) {
    string bombDir;
    cin >> bombDir; cin.ignore();

    if (bombDir == "U") {
      minX = maxX = x;
      maxY = y - 1;
    } else if (bombDir == "UR") {
      minX = x + 1;
      maxY = y - 1;
    } else if (bombDir == "R") {
      minY = maxY = y;
      minX = x + 1;
    } else if (bombDir == "DR") {
      minX = x + 1;
      minY = y + 1;
    } else if (bombDir == "D") {
      minX = maxX = x;
      minY = y + 1;
    } else if (bombDir == "DL") {
      maxX = x - 1;
      minY = y + 1;
    } else if (bombDir == "L") {
      minY = maxY = y;
      maxX = x - 1;
    } else if (bombDir == "UL") {
      maxX = x - 1;
      maxY = y - 1;
    }

    x = (minX + maxX) / 2;
    y = (minY + maxY) / 2;
    // the location where Batman should jump to.
    cout << x << " " << y << endl;
  }
}
